bob/src/api/mod.rs

```diff
@@ -1,2 +1,1069 @@
+            delete(delete_partition::<A>),
+        ),
+        ("/alien", get(alien)),
+        ("/alien/detach", post(detach_alien_partitions::<A>)),
+        ("/alien/dir", get(get_alien_directory::<A>)),
+        (
+            "/vdisks/:vdisk_id/replicas/local/dirs",
+            get(get_local_replica_directories::<A>),
+        ),
+        ("/data/:key", get(get_data::<A>)),
+        ("/data/:key", post(put_data::<A>)),
+    ]
+}
+
+fn data_vdisk_to_scheme(disk: &DataVDisk) -> VDisk {
+    VDisk {
+        id: disk.id(),
+        replicas: collect_replicas_info(disk.replicas()),
+    }
+}
+
+fn collect_disks_info(bob: &BobServer) -> Vec<VDisk> {
+    let mapper = bob.grinder().backend().mapper();
+    mapper.vdisks().values().map(data_vdisk_to_scheme).collect()
+}
+
+#[inline]
+fn get_vdisk_by_id(bob: &BobServer, id: u32) -> Option<VDisk> {
+    find_vdisk(bob, id).map(data_vdisk_to_scheme)
+}
+
+fn find_vdisk(bob: &BobServer, id: u32) -> Option<&DataVDisk> {
+    let mapper = bob.grinder().backend().mapper();
+    mapper.get_vdisk(id)
+}
+
+fn collect_replicas_info(replicas: &[NodeDisk]) -> Vec<Replica> {
+    replicas
+        .iter()
+        .map(|r| Replica {
+            path: r.disk_path().to_owned(),
+            disk: r.disk_name().to_owned(),
+            node: r.node_name().to_owned(),
+        })
+        .collect()
+}
+
+fn not_acceptable_backend() -> StatusExt {
+    let status = StatusExt::new(
+        StatusCode::NOT_ACCEPTABLE,
+        false,
+        "only pearl backend supports partitions".into(),
+    );
+    warn!("{:?}", status);
+    status
+}
+
+// !notice: only finds normal group
+async fn find_group(bob: &BobServer, vdisk_id: u32) -> Result<PearlGroup, StatusExt> {
+    let backend = bob.grinder().backend().inner();
+    debug!("get backend: OK");
+    let (dcs, _) = backend
+        .disk_controllers()
+        .ok_or_else(not_acceptable_backend)?;
+    debug!("get vdisks groups: OK");
+    let needed_dc = dcs
+        .iter()
+        .find(|dc| dc.vdisks().iter().any(|&vd| vd == vdisk_id))
+        .ok_or_else(|| {
+            let err = format!("Disk Controller with vdisk #{} not found", vdisk_id);
+            warn!("{}", err);
+            StatusExt::new(StatusCode::NOT_FOUND, false, err)
+        })?;
+    needed_dc.vdisk_group(vdisk_id).await.map_err(|_| {
+        let err = format!("Disk Controller with vdisk #{} not found", vdisk_id);
+        warn!("{}", err);
+        StatusExt::new(StatusCode::NOT_FOUND, false, err)
+    })
+}
+
+async fn status(Extension(bob): Extension<&BobServer>) -> Json<Node> {
+    let mapper = bob.grinder().backend().mapper();
+    let name = mapper.local_node_name().to_owned();
+    let address = mapper.local_node_address().to_owned();
+    let vdisks = collect_disks_info(bob);
+    let node = Node {
+        name,
+        address,
+        vdisks,
+    };
+    Json(node)
+}
+
+async fn metrics(Extension(bob): Extension<&BobServer>) -> Json<MetricsSnapshotModel> {
```

Comment:
@Boneyan reported, that this method returns an error: `Missing request extension: Extension of type '&bob::server::Server' was not found. Perhaps you forget to add it?`